go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -369,21 +413,55 @@ func newSharedGatewayOpenFlowManager(nodeName, macAddress, gwBridge, gwIntf stri
 			gwIntf, stderr, err)
 	}
 
+	// Validate the numbers of the default OpenFlow rules and the port number of the associated ports, if they are
+	// we assume the OpenFlow rules are intact. Do nothing so ongoing network traffic won't be interrupted;
+	// Otherwise, flush existing OpenFlows and recreate default OpenFlows.
+	//
+	// Note that the numbers of default OpenFlows may be different depends on the IPv4/IPv6 mode or the other
+	// OVN k8s configurations
+
+	// One table 0 and four table 1 flows
```

Comment:
rather than counting flows like this I was hoping we could refactor the entire openflow manager to work with a cache and "atomic" flow replacement like we did with hybrid overlow. @rcarrillocruz was working on this:
https://github.com/ovn-org/ovn-kubernetes/issues/1888

Reply:
relevant code:
https://github.com/openshift/ovn-kubernetes/blob/release-4.6/go-controller/hybrid-overlay/pkg/controller/node_linux.go#L31
https://github.com/openshift/ovn-kubernetes/blob/release-4.6/go-controller/hybrid-overlay/pkg/controller/node_linux.go#L51
https://github.com/openshift/ovn-kubernetes/blob/release-4.6/go-controller/hybrid-overlay/pkg/controller/node_linux.go#L219
https://github.com/openshift/ovn-kubernetes/blob/release-4.6/go-controller/hybrid-overlay/pkg/controller/node_linux.go#L578
https://github.com/openshift/ovn-kubernetes/blob/release-4.6/go-controller/pkg/util/ovs.go#L671

Reply:
@trozet I looked at the hybrid overlay implementation that you pointed out, I think it is great to sync OpenFlows. But I think it can only get all the flows added (included those for the Pods) when watchPods() is called, which means it will also delete the needed flows at first which could interrupt existing traffic.

so in order  to use the suggested openFlow manager cache approach and not to wipe out the existing openFlows rules added for nodePort services which would interrupt the existing traffic, we need to somehow to defer the intialization of openflow to where nodePortWatcher()->syncServices() is called, so that we can also sync the nodePort service flows as well, is that a ok approach?

Thanks